### PR TITLE
feat: add GitHub URL cloning to rally onboard

### LINKED ISSUE #11: Add GitHub URL cloning to rally onboard
## Description
- Extend `lib/onboard.js` for URL variant
- Accept `https://github.com/owner/repo` or `owner/repo` shorthand
- Clone to `~/.rally/projects/<repo>/`
- Skip clone if directory exists
- Then proceed with normal onboard flow

## Error Cases
- Invalid URL format
- Clone fails (network, auth)
- Repo not found

## Acceptance Criteria
- [ ] Clones from full URL
- [ ] Clones from owner/repo shorthand
- [ ] Skips existing clone
- [ ] Proceeds to normal onboard after clone

**Size:** S | **Owner:** Wash | **Depends on:** #10

### PR DESCRIPTION
**Pull request by jsturtevant**
Closes #11

## Summary
Extends `rally onboard` to accept GitHub URLs and `owner/repo` shorthand in addition to local paths.

### What changed
- **`lib/onboard.js`**: Added `parseGithubUrl()` that detects `https://github.com/owner/repo[.git]` and `owner/repo` shorthand. When a URL is detected, the repo is cloned to `~/.rally/projects/<repo>/` (respects `RALLY_HOME`), then the normal onboard flow proceeds. Skips clone if target dir exists.
- **`bin/rally.js`**: Updated command description to reflect URL support.
- **`test/onboard-url.test.js`**: 15 new tests covering URL parsing (all formats, edge cases, null/empty input) and integration (clone skip, normal flow after clone, error on failed clone, local path still works).

### Acceptance Criteria
- [x] Clones from full URL (`https://github.com/owner/repo`)
- [x] Clones from `owner/repo` shorthand
- [x] Skips existing clone
- [x] Proceeds to normal onboard after clone
- [x] Error handling for clone failures
- [x] All 101 tests pass

### CODE REVIEW

File: lib/onboard.js [hunk 1]
```diff
@@ -2,20 +2,83 @@ import { existsSync, mkdirSync, realpathSync, lstatSync, readlinkSync } from 'no
...
+/**
+ * Parse a GitHub URL or owner/repo shorthand into clone metadata.
+ * Returns { owner, repo, cloneUrl } or null if input looks like a local path.
+ */
+export function parseGithubUrl(input) {
+  if (!input) return null;
+
+  // Full URL: https://github.com/owner/repo[.git]
+  const urlMatch = input.match(
+    /^https:\/\/github\.com\/([A-Za-z0-9_.-]+)\/([A-Za-z0-9_.-]+?)(?:\.git)?$/
```
Comment (jsturtevant) [inline lib/onboard.js]: **Minor bug: trailing slash causes silent fallthrough.**

`https://github.com/owner/repo/` doesn't match this regex (the trailing `/` isn't consumed), so it falls through to local path handling and fails with a confusing "Not a git repository" error.

Easy fix — allow an optional trailing slash:
```js
/^https:\/\/github\.com\/([A-Za-z0-9_.-]+)\/([A-Za-z0-9_.-]+?)(?:\.git)?\/?$/
```

Same applies to the shorthand regex on line 29.

File: lib/onboard.js [hunk 2]
```diff
@@ -2,20 +2,83 @@ import { existsSync, mkdirSync, realpathSync, lstatSync, readlinkSync } from 'no
...
+  const input = options.path || process.cwd();
+
+  // Resolve GitHub URL cloning before entering the normal onboard flow
+  const parsed = parseGithubUrl(input);
+  let projectPath;
+
+  if (parsed) {
+    const configDir = getConfigDir();
+    const projectsDir = join(configDir, 'projects');
+    const cloneTarget = join(projectsDir, parsed.repo);
```
Comment (jsturtevant) [inline lib/onboard.js]: **Minor path traversal: `..` is a valid match for `[A-Za-z0-9_.-]+`.**

`parseGithubUrl('owner/..')` returns `{ repo: '..' }`, so `cloneTarget` becomes `join(projectsDir, '..')` which resolves to the config directory itself. Practical risk is low (the clone URL would be invalid, or if the directory exists the `.git` check gates further damage), but it violates the principle of least surprise.

Suggested guard in `parseGithubUrl`, after extracting owner/repo:
```js
if (repo === '.' || repo === '..' || owner === '.' || owner === '..') return null;
```

Worth adding a test case for this too:
```js
test('returns null for dot-dot traversal', () => {
  assert.strictEqual(parseGithubUrl('owner/..'), null);
  assert.strictEqual(parseGithubUrl('../repo'), null);
});
```
